lib/src/main/cpp/utils.h

```diff
@@ -104,6 +104,9 @@ shared_ptr<T> getPtr(JNIEnv* env, jobject thisObj, const char* handleName = "nat
   jclass thisClass = env->GetObjectClass(thisObj);
   jfieldID fidNumber = env->GetFieldID(thisClass, handleName, "J");
   auto handle = reinterpret_cast<shared_ptr<T>*>(env->GetLongField(thisObj, fidNumber));
+    if (handle == nullptr || *handle == nullptr) {
```

Comment:
The `*handle == nullptr` check is redundant - with current code it would work in exactly the same way without that check too.